server/channels/store/sqlstore/post_store.go

```diff
@@ -2683,10 +2684,61 @@ func (s *SqlPostStore) GetParentsForExportAfter(limit int, afterId string, inclu
 			continue
 		}
 
+		flaggedMap, err := s.getFlaggedPostUsernames(rootIds)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to find flagged posts")
+		}
+
+		for _, post := range postsForExport {
+			if usernames, ok := flaggedMap[post.Id]; ok {
+				post.FlaggedBy = slices.Clone(usernames)
+				delete(flaggedMap, post.Id)
+			}
+
+			if len(flaggedMap) == 0 {
+				break
+			}
+		}
+
 		return postsForExport, nil
 	}
 }
 
+func (s *SqlPostStore) getFlaggedPostUsernames(postIds []string) (map[string][]string, error) {
+	aggFn := "string_agg(Users.Username, ', ')"
+	if s.DriverName() == model.DatabaseDriverMysql {
+		aggFn = "GROUP_CONCAT(Users.Username)"
+	}
+
+	flagged := []struct {
+		Id        string
+		FlaggedBy string
+	}{}
+	builder := s.getQueryBuilder().Select("p1.Id, "+aggFn+" as FlaggedBy").
+		FromSelect(sq.Select("*").From("Posts").Where(sq.Eq{"Posts.Id": postIds}), "p1").
+		LeftJoin("Preferences ON p1.Id = Preferences.Name").
+		LeftJoin("Users ON Preferences.UserId = Users.Id").
+		Where("Users.Username IS NOT NULL").
+		GroupBy("p1.Id")
```

Comment:
A couple of questions:

- Is the subquery on Posts substantially better than joining?
- Do we need to select all the columns from Posts?

Reply:
Wondering if we could get this info in the main query itself, avoiding the need for a second query.